src/handshake/server.rs

```diff
@@ -177,7 +168,7 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Server<'a, T> {
 
 		let path = request.path.unwrap_or("/");
 
-		Ok(ClientRequest { ws_key, protocols, path, headers })
+		Ok(ClientRequest { ws_key, protocols, path, headers: request.headers.to_vec() })
```

Comment:
so, the headers is parsed from an array which is `mutally borrowed`  so I had to clone it the headers unfortunately.